### PR TITLE
Remove python2 from CI

### PR DESCRIPTION
**Pull request by larroy**
## Description ##
This PR removes Python2 test from CI.

See:

https://lists.apache.org/list.html?dev@mxnet.apache.org:lte=1M:python2

From now on Python2 compatibility won't be tested and thus we no longer actively support Python2.

### CODE REVIEW

**Comment by larroy** [comment]
@mxnet-label-bot add [pr-work-in-progress]

**Comment by larroy** [comment]
@mxnet-label-bot add [ci]

**Comment by marcoabreu** [comment]
The community has not agreed on deprecating python2 immediately.

**Comment by marcoabreu** [comment]
It's still a few releases down the road, so no need to put in effort at the moment. It might just send the wrong signals.

**Comment by leezu** [comment]
Re-open as per consensus in https://lists.apache.org/thread.html/r3a2db0f22a1680cc56804191446fef2289595798ca19fd17de1ff03e%40%3Cdev.mxnet.apache.org%3E

@larroy would you like to rebase and fix conflicts?

File: ci/docker/runtime_functions.sh [hunk 1]
```diff
@@ -1544,17 +1480,7 @@ nightly_model_backwards_compat_train() {
     ./tests/nightly/model_backwards_compatibility_check/train_mxnet_legacy_models.sh
 }
 
-# Nightly 'MXNet: The Straight Dope' Single-GPU Tests
-nightly_straight_dope_python2_single_gpu_tests() {
-    set -ex
-    cd /work/mxnet/tests/nightly/straight_dope
-    export PYTHONPATH=/work/mxnet/python/
-    export MXNET_TEST_KERNEL=python2
     export DMLC_LOG_STACK_TRACE_DEPTH=10
```
Comment (marcoabreu) [inline ci/docker/runtime_functions.sh]: Is the diff just a bit off or did you miss a few parts?

Comment (szha) [reply ci/docker/runtime_functions.sh]: I think these lines were added after the creation of this PR and thus it's a merge problem.

Comment (larroy) [reply ci/docker/runtime_functions.sh]: thanks for noticing this, fixed.

**Comment by larroy** [comment]
@mxnet-label-bot remove [pr-work-in-progress]

**Comment by larroy** [comment]
@mxnet-label-bot add [pr-awaiting-merge]

File: ci/docker/install/ubuntu_core.sh [hunk 2]
```diff
@@ -21,6 +21,7 @@
 # the whole docker cache for the image
 
 set -ex
+perl -pi -e 's/archive.ubuntu.com/us-west-2.ec2.archive.ubuntu.com/' /etc/apt/sources.list
```
Comment (aaronmarkham) [inline ci/docker/install/ubuntu_core.sh]: What's this going to do?

Comment (larroy) [reply ci/docker/install/ubuntu_core.sh]: use the AWS mirror instead of the default ubuntu one, making docker rebuild significantly faster.

**Comment by larroy** [comment]
> That will cause performance issues if run outside of CI. Maybe not replace? Also, unrelated change

Can be removed before merge. Many CI failures now that require rebuild. Can also have a conditional on being in EC2.

**Comment by marcoabreu** [comment]
Conditional sounds good

**Comment by leezu** [comment]
Static build pipeline failures are addressed by https://github.com/apache/incubator-mxnet/pull/17474

Please review / merge (or include the changes here)

**Comment by larroy** [comment]
> Static build pipeline failures are addressed by #17474
> 
> Please review / merge (or include the changes here)

Thanks, included your commit, but is it related to the failures?

**Comment by leezu** [comment]
Yes. If you check the output of your CI runs, you'll see the staticbuild pipeline consistently fails due to Docker cache invalidation by your PR.

**Comment by larroy** [comment]
> Yes. If you check the output of your CI runs, you'll see the staticbuild pipeline consistently fails due to Docker cache invalidation by your PR.

Could you paste the snippet that you refer to? I'm confused sorry.

**Comment by leezu** [comment]
@larroy let's merge https://github.com/apache/incubator-mxnet/pull/17474 and then you can merge master into your branch.

**Comment by larroy** [comment]
> @larroy let's merge #17474 and then you can merge master into your branch.

Your commit is included anyway.

File: ci/docker/install/ubuntu_core.sh (hunk 2 above)
Comment (marcoabreu) [reply ci/docker/install/ubuntu_core.sh]: Can you make this conditional?

**Pull request merged**